.github/workflows/cwag-workflow.yml

```diff
@@ -0,0 +1,369 @@
+      run: |
+            cd ${MAGMA_ROOT}/cwf/gateway/docker
+            docker-compose -f docker-compose.yml -f docker-compose.override.yml build --parallel
+    - name: tag-push-docker
+      run: |
+          if [ "${GITHUB_BASE_REF}" != "master" ]; then
+            echo "Push only happens for master branch"
+            exit 0
+          fi
+          DOCKER_REGISTRY="$DOCKER_FEG_REGISTRY"
+          DOCKER_USERNAME="$USERNAME"
+          DOCKER_PASSWORD="$PASSWORD"
+          docker login "${DOCKER_REGISTRY}" -u "${DOCKER_USERNAME}" -p "${DOCKER_PASSWORD}"
+          IMAGES="cwag_go|gateway_go|gateway_python|gateway_sessiond|gateway_pipelined"
+          IMAGES_ARRAY=($(echo $IMAGES | tr "|" "\n"))
+          PROJECT=cwf
+          TAG=${GITHUB_SHA:0:8}
+          TAG_LATEST=false
+          function tag_and_push {
+            docker tag "$IMAGE_ID" "${DOCKER_REGISTRY}/$IMAGE:$1"
+            echo "Pushing ${DOCKER_REGISTRY}/$IMAGE:$1"
+            docker push "${DOCKER_REGISTRY}/$IMAGE:$1"
+          }
+          for IMAGE in "${IMAGES_ARRAY[@]}"; do
+            IMAGE_TOSEARCH=$IMAGE
+            if [ ! -z $PROJECT ]; then
+              IMAGE_TOSEARCH="${PROJECT}_${IMAGE}"
+            fi
+            IMAGE_ID=$(docker images "$IMAGE_TOSEARCH:latest" --format "{{.ID}}")
+            tag_and_push "$TAG"
+            if [ "$TAG_LATEST" = true ]; then
+              tag_and_push "latest"
+            fi
+          done
+    - name: persist-githash-version - Set prefix
+      run: |
+          echo "file_prefix=cwag" >> $GITHUB_ENV
+    - name: persist-githash-version - Create version file
+      run: |
+          cd ${MAGMA_ROOT}/circleci
+          mkdir -p versions
+          echo "${GITHUB_SHA:0:8}" > versions/<< ${file_prefix} >>_version
+#      - magma_slack_notify
+
+########## XWFM-DEPLOY ##########
+
+  xwfm-deploy-latest:
+
+    needs: [cwag-deploy, xwfm-test]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Export Magma root path
+      run: |
+          echo "MAGMA_ROOT=${PWD}" >> $GITHUB_ENV
+          echo "DOCKER_REGISTRY=cwf_" >> $GITHUB_ENV
+          echo "tag-latest=true" >> $GITHUB_ENV
+    - name: Build the docker-compose stack
+      run: |
+           docker-compose up -d
+    - name: Build xwf go radius
+      run: |
+            cd ${MAGMA_ROOT}/feg
+            docker build --build-arg BUILD_NUM=${CIRCLE_SHA1:0:8} --tag goradius -f radius/src/Dockerfile ./
+    - name: Load openvswitch kernel module for xwf integ test
+      run: sudo modprobe openvswitch
+    - name: Build xwfm-integ-tests
+      run: |
+            cd ${MAGMA_ROOT}
+            docker build --tag xwfm-integ-tests -f xwf/gateway/integ_tests/gw/Dockerfile ./
+    - name: tag-push-docker
+      run: |
+          if [ "${GITHUB_BASE_REF}" != "master" ]; then
+            echo "Push only happens for master branch"
+            exit 0
+          fi
+          DOCKER_REGISTRY="$DOCKER_FEG_REGISTRY"
```

Comment:
maybe it would be nice to have that step in a script so that we don't bother having the same code multiple times.
I don't think github actions have reusable steps that you can define easily as circleci.